packages/orm/src/QueryParser.ts

```diff
@@ -61,34 +62,73 @@ export interface PrimaryTable {
   table: string;
 }
 
+/**
+ * Joins into 0-1 relations, i.e. children up to a parent.
+ *
+ * Even though `LateralJoinTable` handles the "many children" case, we keep the
+ * `"outer"` join for doing joins to optional "0 or 1" relations, where we don't
+ * want a missing result to mean the primary entity itself is not returned.
+ */
 export interface JoinTable {
   join: "inner" | "outer";
   alias: string;
   table: string;
   col1: string;
   col2: string;
-  distinct?: boolean;
 }
 
-export type ParsedTable = PrimaryTable | JoinTable;
+export interface CrossJoinTable {
+  join: "cross";
+  alias: string;
+  table: string;
+}
+
+export type ParsedTable = PrimaryTable | JoinTable | CrossJoinTable | LateralJoinTable;
+
+/**
+ * Joins into 0-N relations, i.e. parent down to many children.
+ *
+ * It's expect that the `query` will fundamentally be an aggregate query that
+ * counts (for `em.find`s) or rolls up (for JSON preloading) the children into
+ * a single row, such that our top-level query does not need to DISTINCT-away
+ * any duplication that comes from having multiple children rows.
+ */
+export interface LateralJoinTable {
+  join: "lateral";
+  alias: string;
+  /** Used for join dependency tracking. */
+  fromAlias: string;
+  /** Used more for bookkeeping/consistency with other join tables than the query itself. */
+  table: string;
+  /** The subquery that will look for/roll-up N children. */
+  query: ParsedFindQuery;
+}
 
 export interface ParsedOrderBy {
   alias: string;
   column: string;
   order: OrderBy;
 }
 
+export interface ParsedGroupBy {
+  alias: string;
+  column: string;
+}
+
+type ParsedSelect = string | ParsedSelectWithBindings;
+type ParsedSelectWithBindings = { sql: string; bindings: any[]; aliases: string[] };
+
 /** The result of parsing an `em.find` filter. */
 export interface ParsedFindQuery {
-  selects: string[];
+  selects: ParsedSelect[];
   /** The primary table plus any joins. */
   tables: ParsedTable[];
-  /** Any cross lateral joins, where the `joins: string[]` has the full join as raw SQL; currently only for preloading. */
-  lateralJoins?: { joins: string[]; bindings: any[] };
```

Comment:
This is the original `LATERAL JOIN` infra from the JSON preloader, which is getting replaced with `tables` having `LateralJoinTable`s in them.